locals.tf

```diff
@@ -102,4 +102,86 @@ locals {
     ] : [],
     (var.datadog_enabled && var.ecs_launch_type == "EC2") ? module.datadog.volumes : []
   )
+
+  http_tcp_listeners = var.app_type == "tcp-app" ? [
+    for index, portmap in var.port_mappings :
+      {
+        port               = portmap.host_port
+        protocol           = "TCP"
+        target_group_index = index
+      }
+    ] : [
+      {
+        port               = var.http_port
+        protocol           = "HTTP"
+        target_group_index = 0
+      },]
+
+  ecs_service_tcp_port_mappings = [
+    for index, portmap in var.port_mappings :
+      {
+        container_name      = var.name
+        container_port      = portmap.container_port
+        host_port           = portmap.host_port
+        target_group_arn    = length(module.alb[*].target_group_arns) >= 1 ? module.alb[0].target_group_arns[index] : ""
+      }
+    ]
+
+  target_groups_web = [
```

Comment:
Would this mean that target_groups_web will always be present? Do we need to  add a ternary here?

Reply:
@AutomationD it's just local.tf, condition is here https://github.com/hazelops/terraform-aws-ecs-app/blob/818bdeacefd3ee29ecd47c6ef2c63a3498756541/main.tf#L28